packages/lesswrong/lib/sql/TableIndex.ts

```diff
@@ -64,9 +73,19 @@ class TableIndex {
     if (options?.unique !== this.options?.unique) {
       return false;
     }
-    if (options?.partialFilterExpression !== this.options?.partialFilterExpression) {
+    if (JSON.stringify(options?.partialFilterExpression) !== JSON.stringify(this.options?.partialFilterExpression)) {
       return false;
     }
+    try {
+      const collationType = getCollationType(options?.collation);
+      if (collationType !== this.collationType) {
+        return false;
+      }
+    } catch {
```

Comment:
Best practice would be to catch a particular type of error, but I don't feel strongly.

Reply:
In principle I agree, but I'm pretty sure the only other way we could throw an exception in this particular case is if `this` is undefined and if that happens then we've got much bigger problems to worry about.